src/graphorm/index.ts

```diff
@@ -167,6 +169,27 @@ const obj = new GraphORM({
       flags: {
         jsonType: true,
       },
+      userState: {
+        relation: {
+          isMany: false,
+          customRelation: (ctx, parentAlias, childAlias, qb): QueryBuilder => {
+            return qb
+              .where(`${childAlias}."userId" = :userId`, { userId: ctx.userId })
+              .andWhere(`${childAlias}."postId" = "${parentAlias}".id`);
+          },
+        },
+        transform: (value: GQLUserPost, ctx: Context) => {
+          if (!ctx.userId) {
+            return null;
+          }
+
+          if (!value) {
+            return userPostDefaultData;
+          }
```

Comment:
Wanted to achieve that clients can fetch and get valid `userState` object even when `user_post` entry is empty for some user <-> post combination. Think it will be better DX instead of always looking for null types. Interested to see what you think.

Reply:
We can go around it with optional chaining but happy with this implementation as well. We just have to make sure we maintain this default object

Reply:
@rebelchris interested to hear what you think its better here 🙏

Reply:
This is fine with me, as Ido mentioned just important to keep the default in place and up to date.
Perhaps we can create a test case that hardcodes these so we atleast get notified when they don't match?

Reply:
@rebelchris per your comment I looked how this could be more streamlined to avoid double updating, utilised `repo.create` in dee9abe93f4ecb1848fdcf0e86aa25f25684da79, seems cleaner. WDYT?

note if obscure: `repo.create` just creates an object it does not do any db calls

Reply:
Thats great indeed!